src/map/entities/battleentity.cpp

```diff
@@ -546,9 +546,29 @@ uint16 CBattleEntity::GetRangedWeaponRank()
     return 0;
 }
 
+uint8 CBattleEntity::GetMainWeaponHitCount()
+{
+    if (auto* weapon = dynamic_cast<CItemWeapon*>(m_Weapons[SLOT_MAIN]))
+    {
+        return weapon->getHitCount();
+    }
+
+    return 1;
+}
+
+uint8 CBattleEntity::GetSubWeaponHitCount()
+{
+    if (auto* weapon = dynamic_cast<CItemWeapon*>(m_Weapons[SLOT_SUB]))
+    {
+        return weapon->getHitCount();
+    }
+
+    return 1;
+}
+
 /************************************************************************
  *                                                                       *
- *  Изменяем количество TP сущности                                      *
```

Comment:
I think you can remove from the PR the changes to comments as when we rebase ASB to LSB then I think they already have changed these comments there. Thanks!

Reply:
Gotcha, I've reverted the translations

Reply:
> Gotcha, I've reverted the translations

I still see a few comment changes leftover

Reply:
Oh whoops I see; would you like me to revert the few I capitalized as well?

Reply:
Reverted the one I missed and also the ones I capitalized where I also did not change any code.

Reply:
Thanks, would you be able to check this [PR](https://github.com/LandSandBoat/server/pull/4783/files) as well and align your PR as much as possible with regard to things such as function, variable names, and code (without like creating new files or such). This will help later on when we merge in LSB. Sorry for the further request, a lot of moving parts on these kind of changes.

Reply:
I notice that LSB is using from different conventions in code than ASB. e.g. I noticed other areas of the code in ASB using assertions such as : `XI_DEBUG_BREAK_IF(m_PBaseEntity->objtype == TYPE_NPC);` while in LSB they seem to print a warning instead.
Is the rule of thumb to follow LSB? Is that the more "up to date" code base so to speak?

Reply:
> I notice that LSB is using some different conventions in code than ASB. e.g. I noticed other areas of the code in ASB using assertions such as : `XI_DEBUG_BREAK_IF(m_PBaseEntity->objtype == TYPE_NPC);` while in LSB they seem to print a warning instead. Is the rule of thumb to follow LSB? Is that the more "up to date" code base so to speak?

yeah use LSB code in such cases